pypit/par/parset.py

```diff
@@ -0,0 +1,649 @@
+                provided if :attr:`cfg_section` is None or any of the
+                parameters are not also ParSet instances themselves.
+            section_comment (:obj:`str`, optional):
+                The top-level comment for the config section based on
+                this ParSet.
+            section_level (:obj:`int`, optional):
+                The top level of this ParSet.  Used for recursive output
+                of nested ParSets.
+            append (:obj:`bool`, optional):
+                Append this configuration output of this ParSet to the
+                file.  False by default.  If not appending and the file
+                exists, the file is automatically overwritten.
+            quiet (:obj:`bool`, optional):
+                Suppress all standard output from the function.
+            just_lines (:obj:`bool`, optional):
+                Do not write the file.  Just construct the file lines.
+
+        Raises:
+            ValueError:
+                Raised if there are types other than ParSet in the
+                parameter list, :attr:`cfg_section` is None, and no
+                section_name argument was provided.
+        """
+        if not just_lines and os.path.isfile(cfg_file) and not append and not quiet:
+            warnings.warn('Selected configuration file already exists and will be overwritten!')
+
+        config_output = []
+        if numpy.all([ isinstance(d, ParSet) for d in self.data.values() ]):
+            # All the elements are ParSets themselves, so just iterate
+            # through each one
+            for k in self.keys():
+                config_output += ParSet.config_lines(self.data[k], section_name=k,
+                                                     section_comment=self.descr[k],
+                                                     section_level=section_level)
+                config_output += ['']
+        else:
+            if section_name is None and self.cfg_section is None:
+                raise ValueError('No top-level section name available for configuration!')
+
+            _section_name = self.cfg_section if section_name is None else section_name
+            _section_comment = self.cfg_comment if section_comment is None else section_comment
+            config_output += ParSet.config_lines(self, section_name=_section_name,
+                                                 section_comment=_section_comment,
+                                                 section_level=section_level)
+
+        if just_lines:
+            return config_output
+
+        with open(cfg_file, 'a' if append else 'w') as f:
+            f.write('\n'.join(config_output))
+
+
+class ParDatabase(object):
+    """
+    Class used as a list of ParSets in a glorified structured numpy
+    array.
+
+    Very similar to yanny when converted to a numpy array.
+
+    Can be initialized using a list of ParSet objects, or an SDSS
```

Comment:
SDSS?  :)

Reply:
Ha!  ParDatabase actually isn't used in pypit, but I left it in.